variables.tf

```diff
@@ -177,170 +200,372 @@ variable "additional_node_groups" {
-}
+variable "network" {
+  description = <<EOF
+    vpc = {
+      id = Existing vpc id, it will bypass creation by this module.
+      subnets = {
+        private = Existing private subnets.
+        public  = Existing public subnets.
+        pod     = Existing pod subnets.
+      }), {})
+    }), {})
+    network_bits = {
+      public  = Number of network bits to allocate to the public subnet. i.e /27 -> 32 IPs.
+      private = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      pod     = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+    }
+    cidrs = {
+      vpc     = The IPv4 CIDR block for the VPC.
+      public  = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      private = Number of network bits to allocate to the private subnet. i.e /19 -> 8,192 IPs.
+      pod     = The IPv4 CIDR block for the Pod subnets.
+    }
+    use_pod_cidr = Use additional pod CIDR range (ie 100.64.0.0/16) for pod/service networking.
+  EOF
 
-variable "private_subnets" {
-  type        = list(string)
-  description = "Optional list of private subnet ids"
-  default     = null
-}
+  type = object({
+    vpc = optional(object({
+      id = optional(string)
+      subnets = optional(object({
+        private = optional(list(string))
+        public  = optional(list(string))
+        pod     = optional(list(string))
+      }), {})
+    }), {})
+    network_bits = optional(object({
+      public  = optional(number)
+      private = optional(number)
+      pod     = optional(number)
+      }
+    ), {})
+    cidrs = optional(object({
+      vpc = optional(string)
+      pod = optional(string)
+    }), {})
+    use_pod_cidr = optional(bool)
+  })
 
-variable "pod_subnets" {
-  type        = list(string)
-  description = "Optional list of pod subnet ids"
-  default     = null
+  default = {}
 }
 
 variable "bastion" {
+  description = <<EOF
+    ami                      = Ami id. Defaults to latest 'amazon_linux_2' ami.
+    instance_type            = "Instance type."
+    authorized_ssh_ip_ranges = List of CIDR ranges permitted for the bastion ssh access.
+    username                 = Bastion user.
+    install_binaries         = Toggle to install required Domino binaries in the bastion.
+  EOF
   type = object({
-    ami                      = optional(string, null) # default will use the latest 'amazon_linux_2' ami
-    instance_type            = optional(string, "t2.micro")
-    authorized_ssh_ip_ranges = optional(list(string), ["0.0.0.0/0"])
-    username                 = optional(string, null)
-    install_binaries         = optional(bool, false)
+    ami_id                   = optional(string) # default will use the latest 'amazon_linux_2' ami
+    instance_type            = optional(string)
+    authorized_ssh_ip_ranges = optional(list(string))
+    username                 = optional(string)
+    install_binaries         = optional(bool)
```

Comment:
defaults seem odd given we had them now we don't.

Reply:
We still do, but in the submodule.
https://github.com/dominodatalab/terraform-aws-eks/blob/427b677d340da19b41ba65bd67046cba04f85f35/submodules/bastion/variables.tf#L72